src/python/gem5/resources/downloader.py

```diff
@@ -253,6 +254,17 @@ def get_resource(
             gem5_version=gem5_version,
         )
 
+        if (
+            "url" in resource_json
+            and resource_json["url"]
+            and (urlparse(resource_json["url"]).scheme == "file")
+        ):
+            local_path = _file_uri_to_path(resource_json["url"])
+            if os.path.exists(local_path):
+                shutil.copy(local_path, to_path)
+                return
+            raise Exception(f"Could not find file")
+
```

Comment:
I suggest you move this logic down to just before https://github.com/gem5/gem5/blob/f29bfc0640c88a79eb7f94454ce31b3237ec0066/src/python/gem5/resources/downloader.py#L311

The only thing you'll really want to do is:

```py
file_uri_path = _file_uri_to_path(resource_json["url"])
if file_uri_path:
    if not file_uri_path():
        raise Exception("bla bla bla, file not found")
    # Best put a helpful message here saying your copying the file
    # from one locaiton to the other, like what we do with the
    # downloading code.
    shutil.copy(file_uri_path, download_path)
else: 
   # The code that's already there. This will end in this else-block
   # with `print(f"Finished downloading resource '{resource_name}'.")`

# Continue with the rest of the function to handle decompression, untarring, etc.
if run_unzip:
...
```